### PR TITLE
[Kernel] Remove KV masking by performing full bkv fetches in the first 2 steps

### PR DESCRIPTION
**Pull request by yaochengji**
# Description

Thanks to @kyuyeunk for identifying the cost of performing kv masking in the RPA kernel, and also having several attempts for the performance optimization.

# Tests

pytest -s -v tests/kernels/ragged_paged_attention_kernel_v3_hd64_test.py

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

**Review by bythew3i — APPROVED** [review]

File: tests/kernels/ragged_paged_attention_kernel_v3_hd64_test.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def gen_random(shape, dtype):
                     (0, 0),
                     (0, 0),
                 ),
-                constant_values=jnp.nan,
+                constant_values=0,
             ).reshape(
```
Comment (bythew3i) [inline tests/kernels/ragged_paged_attention_kernel_v3_hd64_test.py]: why do we change that?
